core/src/ancestor_hashes_service.rs

```diff
@@ -425,16 +425,21 @@ impl AncestorHashesService {
                     stats.invalid_packets += 1;
                     return None;
                 }
-                if ping.verify() {
-                    stats.ping_count += 1;
-                    if let Ok(pong) = Pong::new(&ping, keypair) {
+                if !ping.verify() {
+                    stats.ping_err_verify_count += 1;
+                    return None;
+                }
+                stats.ping_count += 1;
+                // Respond both with and without domain so that the other node
+                // will accept the response regardless of its upgrade status.
+                // TODO: remove domain = false once cluster is upgraded.
+                for domain in [false, true] {
```

Comment:
don't we wind up with 3x ping-pong traffic like this? double pings, two pongs for each ping?

would a feature gate be preferable?

Reply:
no. ping packets are the same as before. pong packets are double. so overall is 50% more traffic.
ping/pong packets are very infrequent (once every 3 minutes), so I think this should be fine.

The problem with feature gate is that there is no specific slot associated with ping/pong packets. So the sender and responder may not agree if the feature is activated or not. So I think this approach would be safest.